src/families/algorand/OptInFlow/01-SelectToken.js

```diff
@@ -0,0 +1,256 @@
+// @flow
+import invariant from "invariant";
+import React, { useCallback, useState } from "react";
+import {
+  View,
+  StyleSheet,
+  SafeAreaView,
+  FlatList,
+  TouchableOpacity,
+} from "react-native";
+import { Trans, useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+
+import { getMainAccount } from "@ledgerhq/live-common/lib/account/helpers";
+import { getAccountBridge } from "@ledgerhq/live-common/lib/bridge/impl";
+import useBridgeTransaction from "@ledgerhq/live-common/lib/bridge/useBridgeTransaction";
+import { listTokensForCryptoCurrency } from "@ledgerhq/live-common/lib/currencies";
+import { extractTokenId } from "@ledgerhq/live-common/lib/families/algorand/tokens";
+
+import type { TokenCurrency } from "@ledgerhq/live-common/lib/types";
+
+import colors from "../../../colors";
+import { ScreenName } from "../../../const";
+import LText from "../../../components/LText";
+import { accountScreenSelector } from "../../../reducers/accounts";
+import { TrackScreen } from "../../../analytics";
+import FilteredSearchBar from "../../../components/FilteredSearchBar";
+import FirstLetterIcon from "../../../components/FirstLetterIcon";
+import KeyboardView from "../../../components/KeyboardView";
+import InfoIcon from "../../../components/InfoIcon";
+import Info from "../../../icons/Info";
+import BottomModal from "../../../components/BottomModal";
+
+const Row = ({
+  item,
+  onPress,
+  onDisabledPress,
+  disabled,
+}: {
+  item: TokenCurrency,
+  onPress: () => void,
+  onDisabledPress: () => void,
+  disabled: boolean,
+}) => {
+  const tokenId = extractTokenId(item.id);
```

Comment:
mmh this is really weird we have to do this to me. also why are we displaying the "id" on the UI? why is not not part in the name in the first place?

Reply:
we show the name and the Asset id it was in the designs